### PR TITLE
FIX compound attributes value supported in CPr update forwarding

### LINKED ISSUE #1440: Compound value in context provider
Sending a updateContext to CB to a field from a Context Provider, we have found that the value (that is a compound attribute) resent is an empty string

time=2015-10-28T15:09:08.108UTC | lvl=DEBUG | trans=N/A | function=ContextAttribute | comp=Orion | msg=ContextAttribute.cpp[100]: Creating a ContextAttribute: compoundValueP at (nil) for attribute 'op_result' at 0x7f28c000d8a0

**Update to CB: 
```
{
    "contextElements": [
        {
            "id": "xxx",
            "type": "aaaa",
            "isPattern": false,
            "attributes": [
                {
                    "name": "op_result",
                    "type": "string",
                    "value": {
                        "code": "200",
                        "extra": "rt-20%3Brrgb-00FF00%3B"
                    }
                }
            ]
        }
    ],
    "updateAction": "UPDATE"
}
```

**Update generated to the ContextProvier:
 ```
"contextResponses" : [
    {
      "contextElement" : {
        "type" : "aaaa",
        "isPattern" : "false",
        "id" : "xxx",
        "attributes" : [
          {
            "name" : "op_result",
            "type" : "string",
            "value" : ""  
          }
        ]
      },
      "statusCode" : {
        "code" : "200",
        "reasonPhrase" : "OK"
      }
    }
  ]
}
```

### PR DESCRIPTION
**Pull request by fgalan**
Fixes #1440

### CODE REVIEW

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -32,5 +32,5 @@ Add:  Hardening: mongoBackend checking for field existence and proper type to av
 Add:  githash in --version (Issue #1363)
 Add:  Added CLI argument for notification mode. Default mode is 'transient' (previous version used 'permanent' implicitly) (Issue #1370)
 Add:  Added CLI argument for connection memory limit.  Default value is 64Kb (Issue #1384)
-Fix:  Separate semahpore for subscription cache, so that sub cache is independent of mutex policy (Issue #1401)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: This is already "integrated" in another line. Maybe it reapeared again due to some merge/rebase conflict resolution...

NTC

File: src/lib/serviceRoutines/postUpdateContext.cpp [hunk 2]
```diff
@@ -541,8 +541,11 @@ std::string postUpdateContext
       }
       else
       {
-        ceP->contextAttributeVector[aIx]->stringValue = aP->stringValue; // FIXME P10: automacit value -> stringValue change, please review to check if it is safe
-        ceP->contextAttributeVector[aIx]->type  = aP->type;
+        ceP->contextAttributeVector[aIx]->stringValue    = aP->stringValue;
+        ceP->contextAttributeVector[aIx]->numberValue    = aP->numberValue;
+        ceP->contextAttributeVector[aIx]->boolValue      = aP->boolValue;
+        ceP->contextAttributeVector[aIx]->valueType      = aP->valueType;
+        ceP->contextAttributeVector[aIx]->compoundValueP = aP->compoundValueP == NULL ? NULL : aP->compoundValueP->clone();
```
Comment (fgalan) [inline src/lib/serviceRoutines/postUpdateContext.cpp]: clone() means new memory being allocated... I'm not sure if and explicit release has to be performed or if it is already done as part of the release of the containing ContextAttribute object

Comment (kzangeli) [inline src/lib/serviceRoutines/postUpdateContext.cpp]: It should be part of ContextAttribute::release

**Comment by kzangeli** [comment]
LGTM

File: src/lib/serviceRoutines/postUpdateContext.cpp (hunk 2 above)
Comment (fgalan) [inline src/lib/serviceRoutines/postUpdateContext.cpp]: NTC

Let's valgrind test suite judge this :)

**Pull request merged**